File: core/src/main/java/com/sequenceiq/cloudbreak/service/messages/CloudbreakMessagesService.java
package com.sequenceiq.cloudbreak.service.messages;

import org.springframework.context.MessageSource;
import org.springframework.stereotype.Component;

import javax.inject.Inject;
import java.util.Collection;
import java.util.Locale;

/**
 * Wraps a message source used by the cloudbreak core.
 * It provides defaults and extends the interface with application specific methods if any
 */
@Component
public class CloudbreakMessagesService {

    @Inject
    private MessageSource messageSource;

    public String getMessage(String code) {
        return messageSource.getMessage(code, null, Locale.getDefault());
    }

    public String getMessage(String code, Collection<?> args) {
        return messageSource.getMessage(code, args.toArray(), Locale.getDefault());
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/messages/CloudbreakMessagesService.kt
package com.sequenceiq.cloudbreak.service.messages

import java.util.Locale

import javax.inject.Inject

import org.springframework.context.MessageSource
import org.springframework.stereotype.Component

/**
 * Wraps a message source used by the cloudbreak core.
 * It provides defaults and extends the interface with application specific methods if any
 */
@Component
class CloudbreakMessagesService {

    @Inject
    private val messageSource: MessageSource? = null

    fun getMessage(code: String): String {
        return messageSource!!.getMessage(code, null, Locale.getDefault())
    }

    fun getMessage(code: String, args: List<Any>): String {
        return messageSource!!.getMessage(code, args.toArray(), Locale.getDefault())
    }
}
